### DISCUSSION TITLE
GitHub actions not terminating on error

### DISCUSSION THREAD

**Discussion by heldersepu** [community/community#26626] [Actions]
<p>We added a syntax check to our pipeline:</p>
<pre><code class="lang-auto">    - name: JS code syntax check
      run: |
        cd functions;
        find . -name "*.js" -exec node --check '{}' \;
</code></pre>
<p>It finds errors but the deployment continues instead of exiting there…<br>
no clue why…<br>
I tested locally and the <code>node --check</code> outputs to the stderr as it should</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/7/f79eda1aa9e20776d62c5ec2c9cf0364163ef5ad.png" data-download-href="/uploads/short-url/zkygZa3ZKAjnKgVA6rKx7RyJwZf.png?dl=1" title="Screenshot from 2021-03-28 22-02-08" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/545474/181098752-c923e074-aa11-4693-b696-dc273740f0c9.png" alt="Screenshot from 2021-03-28 22-02-08" data-base62-sha1="zkygZa3ZKAjnKgVA6rKx7RyJwZf" width="486" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f79eda1aa9e20776d62c5ec2c9cf0364163ef5ad_2_486x500.png, https://aws1.discourse-cdn.com/github/original/3X/f/7/f79eda1aa9e20776d62c5ec2c9cf0364163ef5ad.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/f/7/f79eda1aa9e20776d62c5ec2c9cf0364163ef5ad.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/7/f79eda1aa9e20776d62c5ec2c9cf0364163ef5ad_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2021-03-28 22-02-08</span><span class="informations">707×727 45.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<p>The <code>-exec command {} ;</code> form for find is a check, the file will be considered as a match if the command has an exit status of 0. So the find command over all will be successful if there’s at least one file without warnings.</p>
<p><code>-exec command {} +</code> might work better for what you’re trying to do, because with it find will actually exit with non-zero if the command fails. Note however that this form may get multiple files, I’m not familiar with Node so I can’t say if that might be a problem for your use case.</p>

**Comment by heldersepu** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="171233">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>o the find command over all will be successful if there’s at least one file without warnings.</p>
</blockquote>
</aside>
<p>Ye the <code>{} +</code> did not work … I was checking the exit status (<code>echo $?</code>) and you are correct find is not showing 1… I guess will have to redirect the stderr to a file and check it.</p>
<p>Thanks</p>

**Comment by heldersepu** [comment]
<p>For anyone running into a similar issue here is my fix/workaround:</p>
<pre><code class="lang-auto">    - name: JS code syntax check
      run: |
        cd functions;
        find . -name "*.js" -exec node --check '{}' \; 2&gt; node_err.log
        if [ -s node_err.log ]; then
          cat node_err.log;
          rm node_err.log;
          exit 1;
        fi
</code></pre>
<p>The pipeline now breaks as it should</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/4/d425a7dc8995e4e884b92a82a26b17ed7c0df463.png" data-download-href="/uploads/short-url/ugJTQmEzUBdfhdcPhqR7q2uIG67.png?dl=1" title="Screenshot from 2021-03-29 11-27-16" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/545474/181098756-96300eaf-8d07-477f-9dbc-43f979701655.png" alt="Screenshot from 2021-03-29 11-27-16" data-base62-sha1="ugJTQmEzUBdfhdcPhqR7q2uIG67" width="379" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d425a7dc8995e4e884b92a82a26b17ed7c0df463_2_379x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d425a7dc8995e4e884b92a82a26b17ed7c0df463_2_568x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/4/d425a7dc8995e4e884b92a82a26b17ed7c0df463.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/4/d425a7dc8995e4e884b92a82a26b17ed7c0df463_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2021-03-29 11-27-16</span><span class="informations">605×797 46.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
